src/scalaDebugger.ts

```diff
@@ -46,15 +50,161 @@ export async function start(
-    return [];
+    const mainClassPick = "Main Class";
+    const testClassPick = "Test Suite";
+
+    return vscode.window
+      .showQuickPick([mainClassPick, testClassPick], {
+        placeHolder: "Pick kind of class to debug",
+      })
+      .then((result) => {
+        if (result === mainClassPick) {
+          return this.provideDebugMainClassConfiguration().then(
+            (config) => [config],
+            (_) => []
+          );
+        } else if (result === testClassPick) {
+          return this.provideDebugTestClassConfiguration().then(
+            (config) => [config],
+            (_) => []
+          );
+        }
+        return [];
+      });
   }
 
   resolveDebugConfiguration(
     _folder: WorkspaceFolder | undefined,
     debugConfiguration: DebugConfiguration
   ): ProviderResult<DebugConfiguration> {
+    if (debugConfiguration.type === undefined) return null;
     return debugConfiguration;
   }
+
+  private provideDebugMainClassConfiguration(): Thenable<DebugConfiguration> {
+    return this.askForBuildTarget().then((buildTarget) =>
+      this.askForClassName().then((className) =>
+        this.askForArguments().then((args) =>
+          this.askForConfigurationName(className).then((name) => {
+            const result: DebugConfiguration = {
+              type: configurationType,
+              name: name,
+              request: launchRequestType,
+              mainClass: className,
+              buildTarget: buildTarget,
+              args: args,
+            };
+            return result;
+          })
+        )
+      )
+    );
+  }
+
+  private provideDebugTestClassConfiguration(): Thenable<DebugConfiguration> {
+    return this.askForBuildTarget().then((buildTarget) =>
+      this.askForClassName().then((className) =>
+        this.askForConfigurationName(className).then((name) => {
+          const result: DebugConfiguration = {
+            type: configurationType,
+            name: name,
+            request: launchRequestType,
+            testClass: className,
+            buildTarget: buildTarget,
+          };
+          return result;
+        })
+      )
+    );
+  }
+
+  private askForBuildTarget(): Thenable<string | undefined> {
+    return vscode.window
+      .showInputBox({
+        prompt: "Enter name of the build target",
+        placeHolder: "Optional, you can leave it empty",
+      })
+      .then((buildTarget) => {
+        if (buildTarget === undefined) {
+          return Promise.reject();
+        } else if (buildTarget === "") {
+          return undefined;
+        } else {
+          return buildTarget;
+        }
+      });
+  }
+
+  private askForClassName(): Thenable<string> {
+    return vscode.window
+      .showInputBox({
+        prompt: "Enter the name of the class to debug",
+        placeHolder: "<package>.<Class>",
+      })
+      .then((name) => {
+        if (name === undefined) return Promise.reject();
+        return name;
```

Comment:
you can use the new fancy `??` operator if you want:


```suggestion
        return name ?? Promise.reject();
```

Reply:
done, thanks.